lib/timber/events/custom_event.ex

```diff
@@ -15,29 +15,52 @@ defmodule Timber.Events.CustomEvent do
   """
 
   @type t :: %__MODULE__{
-    name: String.t | nil,
-    data: map | nil
+    name: String.t,
+    data: map | nil,
+    time_ms: float() | nil
   }
 
+  @enforce_keys [:name]
   defstruct [
     :data,
-    :name
+    :name,
+    :time_ms
   ]
 
   @doc ~S"""
   Creates a new custom event
 
-  ```
-  event_data = %{customer_id: "xiaus1934", amount: 1900, currency: "USD"}
-  event = CustomEvent.new(name: :payment_received, data: event_data)
-
-  Logger.info("Received payment", timber_event: event)
-  ```
-
   Note: You cannot use tuples in your data structure. Trying to include them
   will cause an encoding error.
+
+  ## Examples
+
+  Basic example:
+
+    iex> require Logger
+    iex> event_data = %{customer_id: "xiaus1934", amount: 1900, currency: "USD"}
+    iex> event = Timber.event(name: :payment_received, data: event_data)
+    iex> Logger.info("Received payment", timber_event: event)
+
+  With timing:
+
+    iex> require Logger
+    iex> timer = Timber.start_timing()3
+    iex> # ... code to time ...
+    iex> event_data = %{customer_id: "xiaus1934", amount: 1900, currency: "USD"}
+    iex> event = CustomEvent.new(name: :payment_received, data: event_data, timer: timer)
+    iex> Logger.info("Received payment", timber_event: event)
+
   """
   @spec new(Keyword.t) :: t
+  def new([timer: timer] = opts) do
```

Comment:
Don't try to pattern match here. Keyword lists are ordered, so if it's not the first element, it will blow up. Instead, you'll need to merge this with the standard `new/1` function and just operate on whether it has a `:timer` key in `opts` or not.